crates/ra_prof/src/memory_usage.rs

```diff
@@ -3,36 +3,43 @@ use std::fmt;
 
 use cfg_if::cfg_if;
 
+#[derive(Copy, Clone)]
 pub struct MemoryUsage {
     pub allocated: Bytes,
-    pub resident: Bytes,
+}
+
+impl fmt::Display for MemoryUsage {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", self.allocated)
+    }
+}
+
+impl std::ops::Sub for MemoryUsage {
+    type Output = MemoryUsage;
+    fn sub(self, rhs: MemoryUsage) -> MemoryUsage {
+        MemoryUsage { allocated: self.allocated - rhs.allocated }
+    }
 }
 
 impl MemoryUsage {
     pub fn current() -> MemoryUsage {
         cfg_if! {
             if #[cfg(target_os = "linux")] {
                 // Note: This is incredibly slow.
-                let alloc = unsafe { libc::mallinfo() }.uordblks as u32 as usize;
-                MemoryUsage { allocated: Bytes(alloc), resident: Bytes(0) }
+                let alloc = unsafe { libc::mallinfo() }.uordblks as u32 as isize;
```

Comment:
```c
           struct mallinfo {
               int arena;     /* Non-mmapped space allocated (bytes) */
               int ordblks;   /* Number of free chunks */
               int smblks;    /* Number of free fastbin blocks */
               int hblks;     /* Number of mmapped regions */
               int hblkhd;    /* Space allocated in mmapped regions (bytes) */
               int usmblks;   /* Maximum total allocated space (bytes) */
               int fsmblks;   /* Space in freed fastbin blocks (bytes) */
               int uordblks;  /* Total allocated space (bytes) */
               int fordblks;  /* Total free space (bytes) */
               int keepcost;  /* Top-most, releasable space (bytes) */
           };
```

I guess we could cast to `i32` instead of `u32` (or to `isize` directly?)